stm32-p103 shell command linker script or hook maybe wrong

```
➜  pikoRT git:(master) ✗ arm-none-eabi-nm piko.elf | grep 0063
00006360 T __rodata_start__  
00006330 t shell_cat         
00006360 T __shell_cmd_end__ 
0000632e T __shell_cmd_start__                            
00006338 t shell_echo        
00006340 t shell_exit        
00006348 t shell_halt        
00006350 t shell_ls          
00006358 t shell_reboot      
00006312 T strchr            
0000632e T __text_end__      
```
It should be something like this IMO:
```
0000632e T __shell_cmd_start__                            
00006338 t shell_echo        
00006340 t shell_exit        
00006348 t shell_halt        
00006350 t shell_ls          
00006358 t shell_reboot   
xxxxxxxxx T __shell_cmd_end__ 
00006312 T strchr            
0000632e T __text_end__      

```

This cause a fatal error in stm32-p103 when executing:
```
$ ls
qemu: fatal: Trying to execute code outside RAM or ROM at 0x5c690000                                                  

R00=00000001 R01=20008f78 R02=20008f78 R03=5c690000        
R04=6bd80000 R05=00000000 R06=00000000 R07=20008f70        
R08=00000000 R09=00000000 R10=00000000 R11=00000000        
R12=00000000 R13=20008f70 R14=00005f19 R15=5c690000        
PSR=00000153 ---- A svc32    
FPSCR: 00000000              
make: *** [target/stm32p103/Makefile.rules:31: dbg] Aborted (core dumped)                                             
```

--- a/piko.lds.S
+++ b/piko.lds.S
@@ -20,6 +20,7 @@ SECTIONS
         PROVIDE(__text_start__ = .);
         *(.text*)
         PROVIDE(__text_end__ = .);
+        . = ALIGN(8);
         PROVIDE(__shell_cmd_start__ = .);
         KEEP(*(.shell_cmd*))
         PROVIDE(__shell_cmd_end__ = .);

